app/src/routes/api/v1/layer.router.js

```diff
@@ -83,10 +83,39 @@ class Layer {
       ctx.throw(403, 'Only team managers can create team layers.');
     }
   }
+
+  static async patchLayer(ctx) {
+    logger.info('Patch layer by id');
+    const layerId = ctx.params.layerId;
+    const body = ctx.request.body;
+    let layer = null;
+    try {
+      layer = await LayerModel.findOne({ _id: layerId });
+      if (!layer) ctx.throw(404, 'Layer not found');
+    } catch (e) {
+      logger.error(e);
+      ctx.throw(500, 'Layer retrieval failed.');
+    }
+    let team = null;
+    if (layer.owner.type === LayerService.type.TEAM) {
+      try {
+        team = await TeamService.getTeam(layer.owner.id);
+      } catch (e) {
+        logger.error(e);
+        ctx.throw(500, 'Team retrieval failed');
+      }
+    }
+    const enabled = LayerService.getEnabled(layer, body, team);
+    const isPublic = LayerService.updateIsPublic(layer, body);
+    layer = Object.assign(layer, body, { isPublic, enabled });
+    await layer.save();
```

Comment:
Why not include a try and catch here? just to prevent wrong api responses and notify users the real status of the layer.